### PR TITLE
Refactor HiveTestUtils#SESSION to class Session

### PR DESCRIPTION
**Pull request by fredlacis**
As suggested by @shixuan-fan on issue #12367, I refactored the constant SESSION to be a Session class instead of a ConnectorSession.
I initialized it with the same options as the previously used ConnectorSession using the testingSessionBuilder from the TestingSession class and changed all its use cases to SESSION.toConnectorSession().

Test plan
- To be honest, as a beginner, I didn't know exactly how to test it, I was hoping I could get some help from de CI/CD. I'm open to suggestions as well.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Review by shixuan-fan — APPROVED** [review]
LGTM

**Comment by shixuan-fan** [comment]
This is a test class so we should be good if it could pass the existing tests :)

**Comment by fredlacis** [comment]
@shixuan-fan I'm having some trouble debugging this error. Do you have any idea what could cause that? Or any suggestions on how to debug it more deeply?

**Comment by shixuan-fan** [comment]
@fredlacis It might be a maven problem so I'll restart the test flow. One thing you could try is to see if you could run `TestHiveClient` locally from what the error message says.

```
Error:  An error occurred while instantiating class com.facebook.presto.hive.TestHiveClient: null
Error:  	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:658)
Error:  	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:533)
Error:  	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:278)
Error:  	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:244)
Error:  	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1194)
Error:  	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1022)
Error:  	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:868)
Error:  	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
Error:  	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
Error:  	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
Error:  	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
Error:  	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
Error:  	at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call(MultiThreadedBuilder.java:190)
Error:  	at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call(MultiThreadedBuilder.java:186)
Error:  	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
Error:  	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
Error:  	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
Error:  	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
Error:  	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
Error:  	at java.lang.Thread.run(Thread.java:748)
Error:  -> [Help 1]
```

**Comment by fredlacis** [comment]
@shixuan-fan Locally, I get the following error:

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.0:test (default-test) on project presto-hive: There are test failures.

And then the same error that you reported, but on TestHiveClientInMemoryMetastoreWithFilterPushdown.

I thought it could be a problem with my development environment, Maven, or even the Java Version. But it looks like it isn't.

Running it on Debug Mode made me realize that the error occurs even before the initialization of the SESSION constant.

Do you have any tips on how to debug it?

File: presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveClient.java [hunk 1]
```diff
@@ -361,7 +361,7 @@
             .build();
 
     private static final MaterializedResult CREATE_TABLE_DATA =
-            MaterializedResult.resultBuilder(SESSION, BIGINT, createUnboundedVarcharType(), TINYINT, SMALLINT, INTEGER, BIGINT, REAL, DOUBLE, BOOLEAN, ARRAY_TYPE, MAP_TYPE, ROW_TYPE)
+            MaterializedResult.resultBuilder(SESSION.toConnectorSession(), BIGINT, createUnboundedVarcharType(), TINYINT, SMALLINT, INTEGER, BIGINT, REAL, DOUBLE, BOOLEAN, ARRAY_TYPE, MAP_TYPE, ROW_TYPE)
```
Comment (shixuan-fan) [inline presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveClient.java:364]: I think we would need to pass a `ConnectorId` here to make sure we get the right connector session.

**Review by shixuan-fan — COMMENTED** [review]
I suspect that the reason we are seeing `null` failure is that the generated connector session does not have connector information. 

To make it easier, how about we also create a `CONNECTOR_SESSION` in the util class, and make sure we pass the `ConnectorId` when creating connector session?

**Comment by fredlacis** [comment]
> I suspect that the reason we are seeing `null` failure is that the generated connector session does not have connector information.
> 
> To make it easier, how about we also create a `CONNECTOR_SESSION` in the util class, and make sure we pass the `ConnectorId` when creating connector session?

Should I create this `CONNECTOR_SESSION` from the `SESSION`, or just bring back the original `SESSION` and rename it to `CONNECTOR_SESSION`?

**Comment by shixuan-fan** [comment]
I think creating `CONNECTOR_SESSION` from `SESSION` sounds more reasonable if possible.

**Comment by fredlacis** [comment]
@shixuan-fan Before creating the `CONNECTOR_SESSION` I decided to try to change the `toConnectorSession` method to include the `ConnectorId`. I couldn't make it work, but I think this could be a solution, what do you think?

What I tried to do for now was:
```Java
public ConnectorSession toConnectorSession()
    {
        Map.Entry<ConnectorId, Map<String, String>> propertiesEntry = connectorProperties.entrySet().iterator().next();
        ConnectorId connectorId = propertiesEntry.getKey();
        Map<String, String> properties = propertiesEntry.getValue();
        String presentCatalog = catalog.isPresent() ? catalog.get() : "";
        return new FullConnectorSession(this, identity.toConnectorIdentity(), properties, connectorId, presentCatalog, sessionPropertyManager);
    }
```

**Comment by fredlacis** [comment]
@shixuan-fan I'm not sure where to get the `ConnectorId` from when initializing the `CONNECTOR_SESSION` with the `toConnectorSession(ConnectorId connectorId)` method.

**Comment by shixuan-fan** [comment]
@fredlacis I think the easiest way to test is to create a psvm (`public static void main(String[] args)`) in your code, and just try to create a connector session out of the session because it is the core change.

Here is my suggestions on some basic testing:

1. Use `testSessionBuilder`, which gives default value to most of the required field.
2. `HiveSessionProperties` is not system properties, so instead of creating it in constructor, you should use `addConnectorSessionProperties` instead.

You could try psvm and see if you would hit any new blockers.

**Pull request closed without merging**